Add JIT post-restore cmdline tests

dsouzai opened the pull request:
* Add external messages when JIT Compilation, AOT Compilation, and Method Invalidation occurs due to the post restore options processing.
* Add JIT post-restore tests

dsouzai commented:
@ymanton @tajila could you please review?

ymanton reviewed (approved):
LGTM.

tajila commented:
jenkins test sanity plinuxcriu jdk17

tajila commented on test/functional/cmdLineTests/criu/criu_jitPostRestore.xml:
```diff
@@ -0,0 +1,80 @@
+		<output type="success" caseSensitive="no" regex="no">Killed</output>
+		<output type="required" caseSensitive="yes" regex="no">Pre-checkpoint</output>
+		<output type="success" caseSensitive="yes" regex="no">Post-checkpoint</output>
+		<output type="failure" caseSensitive="yes" regex="no">CRIU is not enabled</output>
```
we've been adding the following to all test because of intermittent failures when criu cant get the thread/processIDs on restore

```
    <!-- If CRIU can't acquire the original thread IDs, this test will fail. Nothing can be done about this failure. -->
    <output type="success" caseSensitive="yes" regex="no">Thread pid mismatch</output>
    <output type="success" caseSensitive="yes" regex="no">do not match expected</output>
    <output type="success" caseSensitive="yes" regex="no">Unable to create a thread:</output>
    <output type="failure" caseSensitive="yes" regex="no">TEST FAILED</output>
    <!-- In the past, the failure below was caused by an issue where CRIU can't be found on the PATH. -->
    <output type="failure" caseSensitive="yes" regex="no">Could not dump the JVM processes, err=-70</output>
```

dsouzai commented:
The three failing tests are
```
[2023-04-10T16:47:52.592Z] FAILED test targets:
[2023-04-10T16:47:52.592Z] 	cmdLineTester_criu_nonPortableRestore_4 
[2023-04-10T16:47:52.592Z] 	cmdLineTester_criu_nonPortableRestore_Xtrace_outputFile_4 
[2023-04-10T16:47:52.592Z] 	cmdLineTester_criu_nonPortableRestoreJDK11Up_0 
```
none of which are news ones I added; the ones I added all passed. Additionally, the output does not have any of the JIT messages I added; these seem to be the same failures as seen in the PR builds in https://github.com/eclipse-openj9/openj9/pull/17132#issuecomment-1503474403.

dsouzai commented:
@tajila good for review again.
